src/main/java/org/simplify4u/plugins/ArtifactResolver.java

```diff
@@ -118,29 +121,31 @@ private ArtifactRepositoryPolicy policyIgnoreCheckSum(ArtifactRepositoryPolicy p
      * @return Returns set of all artifacts whose signature needs to be verified.
      */
     Set<Artifact> resolveProjectArtifacts(MavenProject project, Configuration config) throws MojoExecutionException {
-        final LinkedHashSet<Artifact> allArtifacts = new LinkedHashSet<>(
-                resolveArtifacts(project.getArtifacts(), config.dependencyFilter, config.verifyPomFiles));
+        final LinkedHashSet<Artifact> allArtifacts = new LinkedHashSet<>(resolveArtifacts(project.getArtifacts(),
+                config.dependencyFilter, null, config.verifyPomFiles, false));
         if (config.verifyPlugins) {
             allArtifacts.addAll(resolveArtifacts(project.getPluginArtifacts(), config.pluginFilter,
-                    config.verifyPomFiles));
+                    config.dependencyFilter, config.verifyPomFiles, config.verifyPluginDependencies));
+            // Make an exception to banning deprecated methods, as there is no suitable alternative to accessing
+            // reporting plugins.
+            //noinspection deprecation
             allArtifacts.addAll(resolveArtifacts(project.getReportArtifacts(), config.pluginFilter,
```

Comment:
pleass add @SuppressWarnings( {"deprecation", "java:S1874"} ) to method declaration